aten/src/ATen/native/mkldnn/xpu/FusionUtils.cpp

```diff
@@ -0,0 +1,79 @@
+#include <ATen/native/mkldnn/xpu/FusionUtils.h>
+
+using namespace at::native::onednn;
+
+namespace at::native::xpu {
+
+onednn::Attr unary_attr_with_arg(
+    std::string_view unary,
+    torch::List<std::optional<at::Scalar>> scalars =
+        torch::List<std::optional<at::Scalar>>(),
+    std::optional<std::string_view> algorithm = std::nullopt,
+    onednn::Attr attr = Attr()) {
+  if (unary == "hardswish") {
+    return attr.append_post_eltwise(
+        1.0f, 1.f / 6.f, 1.f / 2.f, attr.kind_with_hardswish);
+  } else if (unary == "silu") {
+    return attr.append_post_eltwise(1.0f, 1.0f, 0.0f, attr.kind_with_swish);
+  } else if (unary == "leaky_relu") {
+    auto alpha = scalars[0].get().toOptional<at::Scalar>().value().to<float>();
+    return attr.append_post_eltwise(1.0, alpha, 0.f, attr.kind_with_relu);
+  } else if (unary == "hardtanh") {
+    auto alpha = scalars[0].get().toOptional<at::Scalar>().value().to<float>();
+    auto beta = scalars[1].get().toOptional<at::Scalar>().value().to<float>();
+    return attr.append_post_eltwise(1.0f, alpha, beta, attr.kind_with_clip);
+  } else if (unary == "gelu") {
+    enum dnnl::algorithm gelu_type;
+    if (algorithm.value() == "none") {
```

Comment:
We should detect `algorithm.has_value()` before fetching its value, otherwise it will crash.

Reply:
add TORCH_CHECK for avoid issue, thanks.